specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "./azure_chat_extensions.tsp";
+import "./common.tsp";
+import "./functions.tsp";
+import "./tools.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "External API shape is defined in OpenAPI 3.0 as oneOf."
+alias ChatMessageContent = string | ChatMessageContentItem[];
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a structured content item within a chat message.")
+@discriminator("type")
+model ChatMessageContentItem {
+  @doc("The discriminated object type.")
+  type: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A structured chat content item containing plain text.")
+model ChatMessageTextContentItem extends ChatMessageContentItem {
+  @doc("The discriminated object type: always 'text' for this type.")
+  type: "text";
+
+  @doc("The content of the message.")
+  text: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A representation of the possible image detail levels for image-based chat completions message content.")
+enum ChatMessageImageDetailLevel {
+  @doc("Specifies that the model should determine which detail level to apply using heuristics like image size.")
+  auto: "auto",
+
+  @doc("""
+  Specifies that image evaluation should be constrained to the 'low-res' model that may be faster and consume fewer
+  tokens but may also be less accurate for highly detailed images.
+  """)
+  low: "low",
+
+  @doc("""
+  Specifies that image evaluation should enable the 'high-res' model that may be more accurate for highly detailed
+  images but may also be slower and consume more tokens.
+  """)
+  high: "high",
+}
+
+@added(ServiceApiVersions.v2023_05_15)
```

Comment:
I am not sure this is entirely correct. Did we support this schema from that version? Could be a pasting mistake, just checking :)

Reply:
Not a mistake, but you're correct that it's also not right.

I'm not sure how to represent this entirely correctly in the TypeSpec such that it won't break the emitters. At the top of the file, we have the `alias ChatMessageContent = string | ChatMessageContentItem[]` thing -- in reality, that second part of that union is what needs to be versioned. There are ways to change types with the versioning library but I'm not sure how to apply it to a *union* like this. Worth following up on, but a known, ugly compromise for the moment.